segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -295,6 +312,29 @@ public void append(Append append) {
                 });
     }
 
+    private void updateIndexSegment(Long lastOffset, Append append) {
+        if (!isTransientSegment(append.getSegment()) && !isTransactionSegment(append.getSegment())) {
+            String indexSegmentName = getIndexSegmentName(append.getSegment());
+            WriterState state = this.writerStates.get(Pair.of(indexSegmentName, null));
+            long previousEventCount = state.getEventCountForAppend();
+            long newEventCount = previousEventCount + append.getEventCount();
+            long maxAllowedEventSize = state.getEventSizeForAppend();
+            WriterState updatedState = new WriterState(newEventCount, maxAllowedEventSize);
+            this.writerStates.put(Pair.of(indexSegmentName, null), updatedState);
+            if (lastOffset != null) {
+                appendOnIndexSegment(append.getSegment(), newEventCount, maxAllowedEventSize, lastOffset);
+            }
+        }
+    }
+
+    private void appendOnIndexSegment(String segmentName, long eventCount, long maxAllowedEventSize, long lastOffset) {
+        String indexSegmentName = getIndexSegmentName(segmentName);
+
+        IndexAppend iAppend = new IndexAppend(indexSegmentName, indexAppendProcessor.getIndexAppendBuf(lastOffset, eventCount), eventCount, maxAllowedEventSize);
```

Comment:
Here a method is being called on `indexAppendProcessor` to obtain a parameter to pass to a different method on `indexAppendProcessor`. Instead just create a method there that takes the relevant parameters and have it do this transformation internally.